avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Author: Harish <harisrir@linux.vnet.ibm.com>
+# Copyright: 2016 IBM
+#
+# Based on the code by
+# Author: Plamen Dimitrov
+# Copyright: Intra2net AG 2012
+
+"""
+Utility for taking shapshots from existing Logical volumes
+or creates such.
+
+:param vg_name: Name of the volume group.
+:param lv_name: Name of the logical volume.
+:param lv_size: Size of the logical volume as string in the form "#G"
+        (for example 30G).
+:param lv_snapshot_name: Name of the snapshot with origin the logical
+        volume.
+:param lv_snapshot_size: Size of the snapshot with origin the logical
+        volume also as "#G".
+:param ramdisk_vg_size: Size of the ramdisk virtual group.
+:param ramdisk_basedir: Base directory for the ramdisk sparse file.
+:param ramdisk_sparse_filename: Name of the ramdisk sparse file.
+:param disk: Name of the disk to create volume group. If not provided,
+             creates loop disk and creates volume group on it.
+
+Sample ramdisk params:
+- ramdisk_vg_size = "40000"
+- ramdisk_basedir = "/tmp"
+- ramdisk_sparse_filename = "virtual_hdd"
+
+Sample general params:
+- vg_name='autotest_vg',
+- lv_name='autotest_lv',
+- lv_size='1G',
+- lv_snapshot_name='autotest_sn',
+- lv_snapshot_size='1G'
+The ramdisk volume group size is in MB.
+"""
+
+import logging
+import os
+import re
+import shutil
+import time
+from . import process
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+def get_diskspace(disk):
+    """
+    Get the entire disk space of a given disk
+    :return: size in bytes
+    """
+
+    result = process.system_output('fdisk -l %s' % disk, shell=True)
+    results = result.splitlines()
+    for line in results:
+        if line.startswith('Disk ' + disk):
+            pattern = re.compile(r", (.*?) bytes")
+            space = pattern.findall(line)[0]
+    return space
```

Comment:
`space` could be undefined value. I'd return the value from inside the `for` loop as suggested earlier and raise `ValueError`/`AssertionError` if we get here.